lib/rspec/core/example_group.rb

```diff
@@ -340,6 +340,16 @@ def self.include_examples(name, *args, &block)
         find_and_eval_shared("examples", name, caller.first, *args, &block)
       end
 
+      # Adds an example to the example group
+      def self.add_example example
+        examples << example
+      end
+
+      # Removes an example from the example group
+      def self.remove_example example
+        examples.delete example
+      end
```

Comment:
reset_memoized is invoked before removing and adding examples. The way that I'm verifying the memoized values are successfully reset is by checking that their object ids have changed. If they changed then the object was successfully reset.

The object id is for these internal instance variables, not the example ids. If I just reset them to an empty array (which retains the object id) then the `||=` logic fails.

If there's a better way to reinitialize the memoized values let me know.